onnxruntime/core/providers/webgpu/allocator.cc

```diff
@@ -13,7 +13,12 @@ void* GpuBufferAllocator::Alloc(size_t size) {
     return nullptr;
   }
 
-  auto buffer = context_.BufferManager().Create(size);
+  WGPUBuffer buffer;
+  if (!session_initialized_ && context_.SupportsBufferMapExtendedUsages()) {
+    buffer = context_.BufferManager().CreateUMA(size);
```

Comment:
you may want to specify the buffer usage to include more flags (map read and map write)